### PR TITLE
Feature/merge home my trees

### LINKED ISSUE #1631: Loading all treecounters of followers when loading a single treecounter
**Describe the bug**
When you look a the tree counter all the tree counters the person of this tree counter is following are loaded by the API too triggering a lot of unnecessary API calls and blocking the UI and making the app slow. 
- Are these additional API calls necessary?
- Why are API calls not done asynchronously in the background updating the frontend using observer pattern?

**Is it on a Production or Development Build?**
production

**To Reproduce**
1. Go to a tree counter of someone following other tree counters
4. See API calls made

**Expected behavior**
Load only data needed to be shown.

**Screenshots**
Simulator Screen Shot - iPhone 11 - 2019-11-14 at 09 32 56

**Smartphone (please complete the following information):**
- Device: iPhone X simulator
 - OS: iOS 13.2
 - Plant-for-the-Planet App Version 1.1.1

### PR DESCRIPTION
**Pull request by harshvitra**
### **Changes in Home page of the user - Private Tree Counter:**

- Adding button to Redeem trees and Register Trees 
- Showing Dedicated user 
- Showing user Competitions
- Showing user Projects ( if TPO )
- Merged my trees in home page (Contributions)
- Show active recurrent donations 
- Edit Profile Button
- Sharing personal profile 
- Map in My Trees

Commenting these sections as I still don't have clear understanding of this
- My supporters section

**To do:** 
- [x] fix headers for all main pages

- [x] Name of TPO is not visible in Profile

- [x] Name of user is now not visible in Profile

Fixes #1631

### CODE REVIEW

**Comment by norbertschuler** [comment]
Does this PR in any way replace #1614?

**Comment by harshvitra** [comment]
> Does this PR in any way replace #1614?

The other one is for web and this for the app

**Comment by sagararyal** [comment]
Screen Shot 2020-01-22 at 8 42 19 PM

This part needs to update.
Including getting rid of menu bar at top

**Comment by harshvitra** [comment]
> * [ ]  Expandable, read more block for description

**The description is divided into 3 blocks already -** 

<Text style={styles.footerText}>
        {userProfile.synopsis1}
</Text>

<Text style={styles.footerText}>
        {userProfile.synopsis2}
</Text>

<Text style={styles.linkText}   onPress={() => this._goToURL(userProfile.url)} >
       {userProfile.linkText || i18n.t('label.read_more')}
</Text>

For implementing read more we need to cut the paragraphs and '...' and the read more button, but we need to them combine these paragraphs before we cut them 

Is it okay if we make  one single paragraph with all these 3 ?

**Comment by harshvitra** [comment]
> This part needs to update.

Updated the top part, but commented out the share button for now

**Comment by harshvitra** [comment]
> * Fixes with padding needs to be implemented.

Screenshot_2020-01-24-16-02-35-658_org pftp

File: package.json [hunk 1]
```diff
@@ -49,6 +49,7 @@
     "intl": "1.2.5",
     "jwt-decode": "^2.2.0",
     "lodash": "^4.17.10",
+    "metro": "^0.56.4",
```
Comment (norbertschuler) [inline package.json]: Why did you add the `metro` module?

File: app/components/UserHome/data.json [hunk 2]
```diff
@@ -0,0 +1,2754 @@
+{
+  "id": 10,
+  "type": "individual",
+  "subType": null,
+  "currency": "USD",
+  "fullname": "Sagar Aryal",
```
Comment (norbertschuler) [inline app/components/UserHome/data.json]: What is this file?

File: app/components/UserContributions/Data Model/object.txt [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  "id": 316,
+  "contributionType": "donation",
+  "cardType": "donation",
+  "treeCount": 50,
+  "plantDate": "2019-12-06",
+  "geoLongitude": -77.899634,
+  "geoLatitude": 86.841271,
+  "country": "",
+  "tpoName": "Plant-for-the-Planet",
+  "plantProjectName": "Campeche Rainforest Restoration",
```
Comment (norbertschuler) [inline app/components/UserContributions/Data Model/object.txt]: What is this file? If these are test data, they should not be committed and pushed to our repository.

Comment (harshvitra) [reply app/components/UserContributions/Data Model/object.txt]: I'll remove this file

File: app/components/UserHome/data.json (hunk 2 above)
Comment (harshvitra) [reply app/components/UserHome/data.json]: I'll remove this file

**Comment by harshvitra** [comment]
@sagararyal we can add maps later.

I've removed header here, which also removes it from everywhere else

**Comment by norbertschuler** [comment]
> I've removed header here, which also removes it from everywhere else

Currently the header doesn't work for the iPhone 11 on almost all pages (probably not regarding safe area any more):

Simulator Screen Shot - iPhone 11 - 2020-01-29 at 13 37 41

**Comment by norbertschuler** [comment]
- [ ] Is the side menu is not accessible any more? Then where are the links to access the marked menu items from the side menu located now?
- My Tree Target
- FAQs
- Logout
- Information

Bildschirmfoto 2020-01-29 um 13 45 55

- [ ] Also the information about the email of the user is not available in the app any more as you can not see it editing your user profile.
- [ ] Due to #1631 (loading all tree counter of followers from a treecounter) loading the "Me" page takes very long doing at lot of API calls now.

**Comment by harshvitra** [comment]
From Sagar -

@harshvitra 

I've got two changes and comments:

- [ ] Remove the buttons that says edit profile, and shows profile type.

Screen Shot 2020-01-29 at 11 42 32 AM

- [ ] Use the Gear icon, to show the side menu for now. (we will replace it in a separate change)

Are you planning to fix the save area view with pr on Top Bar?
Screen Shot 2020-01-29 at 11 33 30 AM

**Comment by sagararyal** [comment]
@harshvitra 

- [ ] Register Tree Screen is Missing Back Button

- [ ] Settings (gear icon) add top padding in display with notch (eg, iPhone X)

- [ ] Add Space before Title (My Trees) and the Link in the Profile.

- [ ] Change Language of Share button "Meet Name on Plant-for-the-Planet App _link_"

**Comment by harshvitra** [comment]
> * [ ]  Is the side menu is not accessible any more? Then where are the links to access the marked menu items from the side menu located now?
> * My Tree Target
> * FAQs
> * Logout
> * Information
> 
> Bildschirmfoto 2020-01-29 um 13 45 55
> 
> * [ ]  Also the information about the email of the user is not available in the app any more as you can not see it editing your user profile.
> * [ ]  Due to #1631 (loading all tree counter of followers from a treecounter) loading the "Me" page takes very long doing at lot of API calls now.

@sagararyal can you please check these, I'm unable to replicate them. Maybe there were some issues with network or API ?

**Comment by norbertschuler** [comment]
> > * [ ]   Is the side menu is not accessible any more? Then where are the links to access the marked menu items from the side menu located now?
> @sagararyal can you please check these, I'm unable to replicate them. Maybe there were some issues with network or API ?

@harshvitra I assume your commit https://github.com/Plant-for-the-Planet-org/treecounter-app/pull/1831/commits/1be0318d60cb51dedecb15a4115cc0201e7ab6d8 fixed it if the user now again has access to the side menu. My issue was just I could not find a way to open the side menu.

**Comment by harshvitra** [comment]
> > > * [ ]    Is the side menu is not accessible any more? Then where are the links to access the marked menu items from the side menu located now?
> > >   @sagararyal can you please check these, I'm unable to replicate them. Maybe there were some issues with network or API ?
> 
> @harshvitra I assume your commit [1be0318](https://github.com/Plant-for-the-Planet-org/treecounter-app/commit/1be0318d60cb51dedecb15a4115cc0201e7ab6d8) fixed it if the user now again has access to the side menu. My issue was just I could not find a way to open the side menu.

Understood !!! :)

**Comment by norbertschuler** [comment]
I removed this issue from my above comment https://github.com/Plant-for-the-Planet-org/treecounter-app/pull/1831#issuecomment-579743924 as they are still open and shouldn't be forgotten as the issue with the side menu is resolved:

- [ ] Also the information about the email of the user is not available in the app any more as you can not see it editing your user profile.
- [ ] Due to #1631 (loading all tree counter of followers from a treecounter) loading the "Me" page takes very long doing at lot of API calls now.

**Comment by norbertschuler** [comment]
- [ ] The app is loading with the "Me" menu items selected, but showing the "World" screen (not yet translated) - the "World" items should be selected:

Simulator Screen Shot - iPhone 11 - 2020-02-03 at 10 26 22

**Comment by norbertschuler** [comment]
- [ ] Date and Numbers of the user contributions are not localized yet - if you do it, please use our standard functions for that task as used everywhere else:

Simulator Screen Shot - iPhone 11 - 2020-02-03 at 10 29 39

**Comment by norbertschuler** [comment]
I have found some issues within the navigation:

- [ ] click on "Register Tree" and then using the back arrow crashes the app:
```
TypeError: props.navigation.goBack is not a function. (In 'props.navigation.goBack()',
 'props.navigation.goBack' is undefined)
TouchableOpacity.props.onPress
    HeaderNew.native.js:48:23
createReactClass$argument_0.touchableHandlePress
    TouchableOpacity.js:264:45
```

- [ ] Due to #1631 (loading all tree counter of followers from a treecounter) loading many pages linked from the "Me" page are loading quite slow, e.g. 
- opening an user contribution
- opening the "Information" page from the side menu 
- opening the about you page
- opening the dedicate trees page
(you need to follow a few other treecounters to reproduce it).

File: app/components/UserHome/index.native.js [hunk 4]
```diff
@@ -117,122 +173,368 @@ export default class UserHome extends Component {
...
+                  //       {userProfile.url ? (
+                  //         <Text
+                  //           style={styles.linkText}
+                  //           onPress={() => this._goToURL(userProfile.url)}
+                  //         >
+                  //           {userProfile.linkText || i18n.t('label.read_more')}
+                  //         </Text>
+                  //       ) : null}
+                  //     </View>
+                  //   ) : null
```
Comment (norbertschuler) [inline app/components/UserHome/index.native.js]: If this code is not needed any more, please delete it - what's the purpose of these lines?

File: app/components/UserHome/index.native.js [hunk 5]
```diff
@@ -117,122 +173,368 @@ export default class UserHome extends Component {
...
+                            style={{ height: 8, width: 15 }}
+                            resizeMode={'contain'}
+                          />
+                        </View>
+                        <Text style={styles.readMoreText}>
+                          {this.state.readMore ? i18n.t('label.read_less') : i18n.t('label.read_more')}
+                        </Text>
+                      </View>
+                    </TouchableOpacity>
+                  ) : null} */}
```
Comment (norbertschuler) [inline app/components/UserHome/index.native.js]: Do we still need this commented code?

File: app/components/UserHome/index.native.js [hunk 6]
```diff
@@ -245,3 +547,173 @@ UserHome.propTypes = {
...
+//   } else {
+//     return (
+//       <>
+//         <CountryLoader />
+//         <CountryLoader />
+//         <CountryLoader />
+//       </>
+//     );
+//   }
+// }
```
Comment (norbertschuler) [inline app/components/UserHome/index.native.js]: What's the purpose of this commented code?

File: app/components/UserHome/index.native copy.js [hunk 7]
```diff
@@ -0,0 +1,247 @@
+/* eslint-disable no-underscore-dangle */
```
Comment (norbertschuler) [inline app/components/UserHome/index.native copy.js]: Is the file `app/components/UserHome/index.native copy.js` used? What is its purpose?

File: package.json (hunk 1 above)
Comment (norbertschuler) [reply package.json]: removed the bundle

File: app/components/UserHome/index.native.js (hunk 4 above)
Comment (harshvitra) [reply app/components/UserHome/index.native.js]: Not for now, I'll save it somewhere else

File: app/components/UserHome/index.native copy.js (hunk 7 above)
Comment (harshvitra) [reply app/components/UserHome/index.native copy.js]: Kept it for quick reference all such files will be removed.

File: app/components/UserHome/index.native.js (hunk 6 above)
Comment (harshvitra) [reply app/components/UserHome/index.native.js]: This is the Supporters section, which is rather a new feature and we designed it already. 
So this will be used in the future.

File: app/components/UserHome/index.native.js (hunk 5 above)
Comment (harshvitra) [reply app/components/UserHome/index.native.js]: Not for now, I'll save it somewhere else

**Comment by norbertschuler** [comment]
- [ ] In my iOS simulator navigating between several screens I sometimes get a delayed rendering of the tab navigation or the page title, so the layout "jumps" into place when the missing elements are added - best visible on the "Leaderboard" screen:

[header_movements.mp4.zip](https://github.com/Plant-for-the-Planet-org/treecounter-app/files/4152226/header_movements.mp4.zip)

**Comment by norbertschuler** [comment]
I got three design issues (not all related to this PR on my Huawei Honor 6, Android 6, 1024x1980 screen):

- [ ] very much whitespace on "Me" screen between avatar and tree counter:

Screenshot_2020-02-05-22-37-56

- [ ] quite a lot whitespace on "Redeem" screen:

Screenshot_2020-02-05-22-39-47

- [ ] third tab item is cut for "Leaderboard":

Screenshot_2020-02-05-22-33-54

**Comment by norbertschuler** [comment]
- [ ] I just got the chance to spot the new design on an iPhone 11 simulator and I think a little more top padding would be nice (compared to the padding you added to the bottom navigation):

Bildschirmfoto 2020-02-07 um 09 22 24

**Comment by norbertschuler** [comment]
- [ ] If we change all screens to the new design, how can the user access the old search function (to find tree counters) or do we abandon this function in the native apps?

**Comment by norbertschuler** [comment]
- [ ] on the "Me" screen, you have to press the button to go to the second page of the treecounter twice before it works:

Screenshot_20200207-141402

File: app/components/TreecounterGraphics/Trillion.native.js [hunk 8]
```diff
@@ -462,6 +462,7 @@ class Trillion extends PureComponent {
             <HeaderStatic
               title={i18n.t('label.explore')}
               navigation={this.props.navigation}
+              showSearch={true}
```
[the remark this answers is not in the document]
Comment (harshvitra) [reply app/components/TreecounterGraphics/Trillion.native.js]: I just had a discussion with Sagar and he guided me to have search in these two screens

**Comment by norbertschuler** [comment]
- [ ] the old treecounter screen had a follow and unfollow button, which is missing now:

old_top
new_top

**Comment by norbertschuler** [comment]
- [ ] the new treecounter screen has a support button hiding some content of the treecounter under the button and you can not scroll up any further (compare to old design):

old_bottom
new_bottom

**Comment by norbertschuler** [comment]
- [ ] When you use the app without logging in and then opening the "Me" screen, you see the login screen. After logging in the "Me" screen is displayed, but the bottom navigation has the "World" item selected.

**Comment by norbertschuler** [comment]
- [ ] In the user contribution details page the text says "Planted by ...". In the old version the TPO name was shown in the link opening the treecounter of the TPO. In this new version the project name is shown and if you click on the link, you get back to the home page instead to the public treecounter of the TPO.

**Comment by norbertschuler** [comment]
- [ ] Details of registered trees (measurements) are not shown in the app:

Bildschirmfoto 2020-02-07 um 19 53 48

**Comment by harshvitra** [comment]
> * [ ]  Details of registered trees (measurements) are not shown in the app:

I can see measurements. 
If you can login into staging with Sagar's account and find the contribution 17,070 Oak trees, you'll see the measurements. And there are many other contributions where I can find them

**Comment by norbertschuler** [comment]
> > * [x]   Details of registered trees (measurements) are not shown in the app:
> 
> I can see measurements.
> If you can login into staging with Sagar's account and find the contribution 17,070 Oak trees, you'll see the measurements. And there are many other contributions where I can find them

Today I saw them too:

Screenshot_2020-02-08-12-59-40

**Comment by norbertschuler** [comment]
> * [x]  the old treecounter screen had a follow and unfollow button, which is missing now:
> @norbertschuler this is intentional

Ok, should we make an issue, that in the app now you can not follow anyone anymore?

**Comment by norbertschuler** [comment]
> * [ ]  When you use the app without logging in and then opening the "Me" or the "Competition" screen, you see the login screen. After logging in the "Me" screen is displayed, but the bottom navigation has the "World" item selected.

This also happens when I do the normal login at the beginning of the app and looks like this on staging (no profile image as assets are missing) - showing "Me" screen, but selecte "World" item:

Screenshot_2020-02-08-13-16-17
